go/enclave/enclave.go

```diff
@@ -276,9 +277,40 @@ func (e *enclaveImpl) Nonce(address gethcommon.Address) uint64 {
 	return s.GetNonce(address)
 }
 
-func (e *enclaveImpl) GetTransaction(txHash gethcommon.Hash) (*common.L2Tx, gethcommon.Hash, uint64, uint64, error) {
-	// todo - joel - encrypt
-	return e.storage.GetTransaction(txHash)
+func (e *enclaveImpl) GetTransaction(encryptedParams common.EncryptedParamsGetTxByHash) (common.EncryptedResponseGetTxByHash, error) {
+	hashBytes, err := e.rpcEncryptionManager.DecryptBytes(encryptedParams)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decrypt encrypted RPC request params. Cause: %w", err)
+	}
+	var paramList []string
+	err = json.Unmarshal(hashBytes, &paramList)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshall RPC request params from JSON. Cause: %w", err)
```

Comment:
Typo that I normally ignore but since it's showing up everywhere at the moment I feel duty bound to mention it haha, it's "marshal" and "unmarshal" for the verbs